src/vmshepherd_aws_drivers/aws_iaas_driver.py

```diff
@@ -0,0 +1,101 @@
+from itertools import chain
+import aiobotocore
+from botocore.exceptions import ClientError
+from typing import Dict, List
+from vmshepherd.iaas import AbstractIaasDriver, Vm, VmState
+
+
+class AwsIaaSDriver(AbstractIaasDriver):
+
+    _VM_STATUSES = {
+        VmState.TERMINATED: [
+            'shutting-down',
+            'terminated',
+            'stopping',
+            'stopped'
+        ],
+        VmState.PENDING: ['pending'],
+        VmState.RUNNING: ['running']
+    }
+
+    def __init__(self, config):
+        self.config = config
+
+    async def list_vms(self, preset_name: str) -> List:
+        '''
+        List VMs by preset name
+        :arg present_name: string
+        '''
+        session = aiobotocore.get_session()
+        async with session.create_client('ec2') as client:
+            paginator = client.get_paginator('describe_instances')
+            filters = [
+                {
+                    'Name': 'tag:aws:autoscaling:groupName',
+                    'Values': [preset_name]
+                }
+            ]
+            instances = []
+            async for result in paginator.paginate(
+                PaginationConfig={'PageSize': self.config.get('ec2_page_size', 5)},
+                Filters=filters
+            ):
+                instances = chain(instances, *[vms['Instances'] for vms in result['Reservations']])
+
+        return [self._map_vm_structure(instance) for instance in instances]
+
+    async def create_vm(self):
+        '''
+        NotImplemented - preset is managed by ASG
+        '''
+        pass
```

Comment:
maybe except, what do you think? like
Exception("Not supported")

Reply:
or at least `logging.warning('something tried to explicitly create vm with aws iaas driver')`

Reply:
good point, solved